src/components/inspector/controls/ExposeChildrenControl.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react'
+import { useSelector } from 'react-redux'
+import { Button, Flex, Input, useToast } from '@chakra-ui/core'
+
+import useDispatch from '../../../hooks/useDispatch'
+import { generateComponentId } from '../../../utils/generateId'
+import {
+  getSelectedComponent,
+  getPropByName,
+  getCustomComponents,
+} from '../../../core/selectors/components'
+import { searchRootCustomComponent } from '../../../utils/recursive'
+import babelQueries from '../../../babel-queries/queries'
+import {
+  getCode,
+  getAllPagesCode,
+  getAllComponentsCode,
+} from '../../../core/selectors/code'
+import { getSelectedPage } from '../../../core/selectors/page'
+
+const ExposeChildrenControl: React.FC<{}> = () => {
+  const dispatch = useDispatch()
+  const toast = useToast()
+  const [name, setName] = useState('')
+
+  const component = useSelector(getSelectedComponent)
+  const childrenProp = useSelector(getPropByName('children'))
+  const customComponents = useSelector(getCustomComponents)
+  const code = useSelector(getCode)
+  const selectedPage = useSelector(getSelectedPage)
+  const pagesCode = useSelector(getAllPagesCode)
+  const componentsCode = useSelector(getAllComponentsCode)
+
+  const { id, children } = component
+  const isCustomComponentChild = customComponents[id] !== undefined
+
+  const changeHandler = (e: any) => setName(e.target.value)
+
+  // todo : Needs to be refactored. UnExposePropButton file has the similar code. Try to use a single function that shares both.
+  const clickHandler = () => {
+    let rootCustomParentElement = ''
+
+    if (isCustomComponentChild) {
+      rootCustomParentElement = searchRootCustomComponent(
+        component,
+        customComponents,
+      )
+    }
+    if (childrenProp) {
+      dispatch.components.unexpose('children')
+
```

Comment:
This clickhandler is doing way too many things. and logic which needs a seperate reducer itself.




And every time a key is pressed this handler is being redefined